n3fit/src/evolven3fit_new/eko_utils.py

```diff
@@ -34,20 +36,51 @@ def construct_eko_cards(
     theory = Loader().check_theoryID(theoryID).get_description()
     theory.pop("FNS")
     theory.update(theory_card_dict)
-    theory_card = gen_theory.gen_theory_card(theory["PTO"], theory["Q0"], update=theory)
+    if "nfref" not in theory:
+        theory["nfref"] = 5
+    if "nf0" not in theory:
+        theory["nf0"] = 4
+    #The Legacy function is able to construct a theory card for eko starting from an NNPDF theory
+    legacy_class = runcards.Legacy(theory, {})
+    theory_card = legacy_class.new_theory
     # construct operator card
     q2_grid = utils.generate_q2grid(
         theory["Q0"],
         q_fin,
         q_points,
         {theory["mb"]: theory["kbThr"], theory["mt"]: theory["ktThr"]},
     )
-    op_card = gen_op.gen_op_card(q2_grid, update={"interpolation_xgrid": x_grid})
-    op_card.update(op_card_dict)
+    op_card = default_op_card
+    op_card.update(
+        {
+            "mu0": theory["Q0"],
+            "_mugrid": np.sqrt(q2_grid).tolist(),
+        }
+    )
+    op_card["configs"].update(
+        {
+            "ev_op_iterations": 1,
+            "ev_op_max_order": (1, 0),
+            "evolution_method": "truncated",
+            "inversion_method": "expanded",
+            "n_integration_cores": 1,
+        }
+    )
+    op_card["rotations"]["xgrid"] = x_grid
+
+    def update_key_if_exists(dict, update_dict, key):
+        if key in update_dict:
+            dict[key].update(update_dict[key])
+
+    for key in op_card:
+        if isinstance(op_card[key], dict):
+            update_key_if_exists(op_card, op_card_dict, key)
+
+    op_card = runcards.OperatorCard.from_dict(op_card)
```

Comment:
I find this part very confusing.

You start with an `op_card` which is  `default_op_card` which you are importing from `eko`.
But this is not a card, is a dictionary, which you need to then update and modify in 3 different ways.

Then this becomes an actual only at the end.

I'm particularly confused about lines 71-77 because if I were to enter:
`op_card_dict = {"mu0": 24}` this would get silently ignored because `op_card["mu0"]` is not a dictionary. 

Same if I enter new parameters in `op_card_dict` that are not included in the original `op_card`. Is this expected or is it a bug?

Reply:
I would not say it is a bug but it is something to be improved. I will :)

Reply:
What I meant is whether is something you actually wanted (then I'd a check to tell the user so that is not silent!) or something you overlooked (then it should be added!)

Reply:
I hope I made it at least a bit clearer